### PR TITLE
fix(build): content-type fails when deployed within internal network controls 

### PR DESCRIPTION
**Pull request by m3mike**
## What Changed
 * Moved ML training to container build time (previously at container startup)
 * Added nginx as reverse proxy in Docker
   * Nginx handles static file delivery
   * Embedded nginx configuration in new `tram-nginx` container image to work in docker compose
   * Updated CI to build nginx container with embedded configuration

### Minor
 * Excluded tox files from container image
 * Updated README link to always point to latest release (previously release version was hardcoded)
 * Removed hardcoded Django SECRET_KEY, key is now autogenerated on application startup for session security

## Limitations
 * Model training now done in Dockerfile, meaning model changes will require a new container image. 
 * Still using sqlite, which will be bottleneck if deployed as-is, may want to switch to postgresql depending on use case
 * As Django secret key is now autogenerated on startup, if application is restarted/reloaded, session key will be regenerated and users will need to login again. To go back to the old behavior, set a static `SECRET_KEY` value in the `docker/docker-compose.yml` file.

### CODE REVIEW

File: .github/workflows/build-push-container.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@
 name: Publish Docker image on publish of release
 
 on:
-  release:
-    types: [published]
+ release:
```
Comment (markeaimark) [inline .github/workflows/build-push-container.yml:6]: This looks inconsistent, there was 2 spaces between each tab, now it seems theres one, then two. Might be worth fixing

File: README.md [hunk 2]
```diff
@@ -9,47 +9,49 @@ TRAM enables researchers to test and refine Machine Learning (ML) models for ide
...
   - [Installation](#installation)
   - [Installation Troubleshooting](#installation-troubleshooting)
     - [[97438] Failed to execute script docker-compose](#97438-failed-to-execute-script-docker-compose)
   - [Requirements](#requirements)
-  - [Developer Setup](#developer-setup)
+  - [For Developers](#for-developers)
     - [Makefile Version](#makefile-version)
       - [Useful User Commands](#useful-user-commands)
       - [Useful Dev Commands](#useful-dev-commands)
+    - [Developer Setup](#developer-setup)
```
Comment (markeaimark) [inline README.md:23]: Is this just getting moved? Or was this meant to be For Developers?

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -50,12 +50,9 @@ else:
...
-python3 /tram/src/tram/manage.py pipeline load-training-data
-python3 /tram/src/tram/manage.py pipeline train --model nb
-python3 /tram/src/tram/manage.py pipeline train --model logreg
 nohup python3 /tram/src/tram/manage.py pipeline run --model logreg &
 
 # Run Django on port 8000
-python3 /tram/src/tram/manage.py runserver 0.0.0.0:8000
+# python3 /tram/src/tram/manage.py runserver 0.0.0.0:8000
+
+gunicorn tram.asgi:application -k uvicorn.workers.UvicornWorker -b 0.0.0.0:8000
```
Comment (markeaimark) [inline docker/entrypoint.sh:58]: Would it be better to keep the dependencies slim and use gunicorn only? Or is there some added benefit to using uvicorn with gunicorn?

Comment (m3mike) [reply docker/entrypoint.sh:58]: I went back and forth. Uvicorn adds the async support to django, so I did it out of habit and figured why not. I could go either way though, no real preference. Thoughts?

File: .github/workflows/build-push-container.yml (hunk 1 above)
Comment (m3mike) [reply .github/workflows/build-push-container.yml]: nice catch, surprised the linters didn't catch it, fixed

File: README.md (hunk 2 above)
Comment (m3mike) [reply README.md:23]: just re-orged the readme a bit

File: docker/entrypoint.sh (hunk 3 above)
Comment (markeaimark) [reply docker/entrypoint.sh:58]: Might be simpler to use just one instead of two

Comment (m3mike) [reply docker/entrypoint.sh]: yea, no async code in the app, removed

**Review by markeaimark — APPROVED** [review]

**Pull request merged**